File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/createnew/fragments/progress/state/ProgressStateImpl.java
package ua.meugen.android.pdfcomposer.ui.activities.createnew.fragments.progress.state;

import java.util.List;

import javax.inject.Inject;

import ua.meugen.android.pdfcomposer.model.data.PageContent;
import ua.meugen.android.pdfcomposer.model.utils.CollectionUtils;
import ua.meugen.android.pdfcomposer.ui.activities.base.fragment.state.BaseState;


public class ProgressStateImpl extends BaseState implements ProgressState {

    @Inject
    ProgressStateImpl() {}

    @Override
    public String getName() {
        return bundle.getString(PARAM_NAME);
    }

    @Override
    public void setName(final String name) {
        bundle.putString(PARAM_NAME, name);
    }

    @Override
    public List<PageContent> getPages() {
        return bundle.getParcelableArrayList(PARAM_PAGES);
    }

    @Override
    public void setPages(final List<PageContent> pages) {
        bundle.putParcelableArrayList(PARAM_PAGES, CollectionUtils.toArrayList(pages));
    }
}

File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/createnew/fragments/progress/state/ProgressStateImpl.kt
package ua.meugen.android.pdfcomposer.ui.activities.createnew.fragments.progress.state

import ua.meugen.android.pdfcomposer.model.data.PageContent
import ua.meugen.android.pdfcomposer.model.utils.toArrayList
import ua.meugen.android.pdfcomposer.ui.activities.base.fragment.state.BaseState


class ProgressStateImpl : BaseState(), ProgressState {

    override var name: String?
        get() = bundle!!.getString(ProgressState.PARAM_NAME)
        set(name) = bundle!!.putString(ProgressState.PARAM_NAME, name)

    override var pages: List<PageContent>?
        get() = bundle!!.getParcelableArrayList(ProgressState.PARAM_PAGES)
        set(pages) = bundle!!.putParcelableArrayList(ProgressState.PARAM_PAGES, pages!!.toArrayList)
}
